geolocation/android/src/main/java/com/capacitorjs/plugins/geolocation/GeolocationPlugin.java

```diff
@@ -47,6 +44,68 @@ public void getCurrentPosition(final PluginCall call) {
         }
     }
 
+    /**
+     * Completes the getCurrentPosition plugin call after a permission request
+     * @see #getCurrentPosition(PluginCall)
+     * @param call the plugin call
+     * @param status the results of the permission request
+     */
+    private void completeCurrentPosition(PluginCall call, Map<String, PermissionState> status) {
+        if (status.get("location") == PermissionState.GRANTED) {
+            boolean enableHighAccuracy = call.getBoolean("enableHighAccuracy", false);
+            int timeout = call.getInt("timeout", 10000);
+
+            implementation.sendLocation(
+                enableHighAccuracy,
+                timeout,
+                true,
+                new LocationResultCallback() {
+                    @Override
+                    public void success(Location location) {
+                        call.resolve(getJSObjectForLocation(location));
+                    }
+
+                    @Override
+                    public void error(String message) {
+                        call.reject(message);
+                    }
+                }
+            );
```

Comment:
Can you use `getPosition()` here?

Reply:
Potentially. I noticed it, and the boolean in the `sendLocation` call made here differs from the one in `getPosition()`, and I didn't want to start digging into that improvement here in the scope of permissions refactoring.